src/pages/DetailProduct/index.jsx

```diff
@@ -1,7 +1,262 @@
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { MinusOutlined, PlusOutlined } from '@ant-design/icons';
+import { v4 as uuidv4 } from 'uuid';
+
+import 'react-image-gallery/styles/css/image-gallery.css';
+import ImageGallery from 'react-image-gallery';
+import CommentAndRating from './CommentAndRating';
+
+import {
+  Radio,
+  Spin,
+  Button,
+  Row,
+  Col,
+  Breadcrumb,
+  Rate,
+  Form,
+  InputNumber,
+  Input,
+  Card,
+} from 'antd';
+
+import { ShoppingCartOutlined } from '@ant-design/icons';
+import { Link, useParams } from 'react-router-dom';
+import { formatNumber, openNotificationWithIcon } from '../../helper';
+
+import { getProductDetail } from '../../redux/actions/product.action';
+import './style.scss';
 
 function DetailProduct() {
-  return <p>DetailProduct</p>;
+  const { productDetails, loading } = useSelector(
+    (state) => state.productReducer,
+  );
+  const { userInfo } = useSelector((state) => state.userReducer);
+  const [imageIndex, setImageIndex] = useState(0);
+  const [size, setSize] = useState(0);
+  const [formComment] = Form.useForm();
+
+  const { id } = useParams();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    window.scroll({ left: 0, top: 0, behavior: 'smooth' });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    dispatch(getProductDetail({ id }));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [id]);
+
+  const submitForm = (values) => {
```

Comment:
```suggestion
  const handleAddToCart = (values) => {
```
hoặc addToCartHandler nhé. Để mỗi submitForm thì khi đọc k biết là nó sẽ làm gì đâu